src/main/java/com/amadeus/travel/TripParser.java

```diff
@@ -89,7 +89,7 @@ public TripDetail post(String body) throws ResponseException {
    */
   public TripDetail post(File file) throws ResponseException, IOException {
     // Base64 encode file and create request body
-    String b64Encoded = Base64.encode(Files.readAllBytes(file.toPath()));
+    String b64Encoded = DatatypeConverter.printBase64Binary(Files.readAllBytes(file.toPath()));
```

Comment:
All I can make sure of is that this new encoding method can get the same result as the original base64 encoding method.

Reply:
Hi @xianqiliu, did you try with this method?
https://docs.oracle.com/javase/8/docs/api/java/util/Base64.html

`java.util.Base64`

Reply:
Hello@jabrena, yep I tried that before cause you have already sent me this link last week. But this method requires Java 1.8. Our compile Java is 1.7, and when I tried this method on my IDE, it suggests that I should set my language to 1.8+. So I search for another method that I use in this PR.

Reply:
@jabrena 
The following is the original code from the JAVA SDK repo's build.gradle
```
compileJava {
    sourceCompatibility = "1.7"
    targetCompatibility = "1.7"
}

compileTestJava {
    sourceCompatibility = "1.8"
    targetCompatibility = "1.8"
}
```

Reply:
@jabrena So from my understanding, the methods we are supposed to use must be compatible with Java 1.7. But I don't know if I misunderstand or not.

Reply:
I understand it.
In that case, we will `pause` this PR, because we will review this issue:
https://github.com/amadeus4dev/amadeus-java/issues/137

Reply:
you may be interested in this
[](https://github.com/amadeus4dev/amadeus-java/issues/184#issuecomment-1192926857)